lib/ace/mode/scrypt_highlight_rules.js

```diff
@@ -0,0 +1,719 @@
+            regex: /\b([_$[:alpha:]][_$[:alnum:]]*)(\s*)((?:(?:\[[\w.]+\])+)?)/
+        }],
+        "#numbers": [{
+            token: "constant.numeric.hex.scrypt",
+            regex: /\b0[xX][a-fA-F0-9]+\b/
+        }, {
+            token: "constant.numeric.integer.scrypt",
+            regex: /\b\d+(?:\.\d+)?\b/
+        }],
+        "#bytes": [{
+            token: "string.quoted.single.bytes.scrypt",
+            regex: /\bb'[0-9a-fA-F]*'/
+        }],
+        "#special-vars": [{
+            token: "variable.language.this.scrypt",
+            regex: /\bthis\b/
+        }],
+        "#proerties-and-functions": [{
+            include: "#properties"
+        }, {
+            include: "#static-properties"
+        }, {
+            include: "#functions"
+        }],
+        "#functions": [{
+            token: "meta.method.scrypt",
+            regex: /(?=[\w].*\s+)(?=(?:[^=\/]|\/(?!\/))+\()/,
+            push: [{
+                token: "meta.method.scrypt",
+                regex: /}|(?=;)/,
+                next: "pop"
+            }, {
+                include: "#contract-constructor"
+            }, {
+                include: "#contract-function"
+            }, {
+                defaultToken: "meta.method.scrypt"
+            }]
+        }],
+        "#contract-constructor": [{
+            token: [
+                "storage.type.function.scrypt",
+                "meta.function.scrypt"
+            ],
+            regex: /\b(constructor)(\s*\()/,
+            push: [{
+                token: "meta.function.scrypt",
+                regex: /\)/,
+                next: "pop"
+            }, {
+                include: "#parameters"
+            }, {
+                include: "#parens"
+            }, {
+                include: "#comments"
+            }, {
+                defaultToken: "meta.function.scrypt"
+            }]
+        }, {
+            include: "#function-body"
+        }],
+        "#contract-function": [{
+            include: "#storage-modifiers"
+        }, {
+            token: [
+                "storage.type.function.scrypt",
+                "meta.function.scrypt",
+                "entity.name.function.scrypt",
+                "meta.function.scrypt"
+            ],
+            regex: /\b(function)(\s+)([_$[:alpha:]][_$[:alnum:]]*)(\s*\()/,
+            push: [{
+                token: "meta.function.scrypt",
+                regex: /\)/,
+                next: "pop"
+            }, {
+                include: "#parameters"
+            }, {
+                include: "#parens"
+            }, {
+                include: "#comments"
+            }, {
+                defaultToken: "meta.function.scrypt"
+            }]
+        }, {
+            include: "#function-return"
+        }, {
+            include: "#function-body"
+        }, {
+            include: "#comments"
+        }],
+        "#function-body": [{
+            token: "punctuation.section.method.begin.bracket.curly.scrypt",
```

Comment:
opening and closing brackets need to have the same class otherwise bracket matching won't work